src/core/Song.cpp

```diff
@@ -1420,21 +1420,37 @@ void Song::clearErrors()
 
 void Song::collectError( const QString error )
 {
-	m_errors.append( error );
+	if( !m_errors.contains( error ) )
+	{
+		m_errors[ error ] = 1;
+	} else {
+		m_errors[ error ]++;
+	}
```

Comment:
I believe` } else {` is a style violation. The simplest way to fix this is to move else to a new line. However, assuming it doesn't break the 80 char line length I you could also do:

```suggestion
	if( !m_errors.contains( error ) ) { m_errors[ error ] = 1; }
	else { m_errors[ error ]++; }
```

Which I personally think would be the nicest format.

Reply:
Just checked, the line length in my suggestion is fine (also the max is 120 and not 80, oops).

Reply:
FYI, `m_errors[error]++` always works because `QHash::operator[]` will automatically insert 0 if the key is not in the hashmap. The same holds for STL containers as well.
I'm okay with using an `if`-`else` block, too.

Also, please change `m_errors[ error ]` to `m_errors[error]`.
https://github.com/LMMS/lmms/wiki/Coding-Conventions#braces